### PR TITLE
chore: try not to flap snapshots

### PR DESCRIPTION
**Pull request by pauldambra**
## Problem

The rate limiter's check of instance setting is cached. This means if we call APIs in tests the rate limiter will fire and cache the instance setting, but if we don't it, erm, won't.

So, where we have tests that count the number of SQL queries made during a statement then the number can change depending on if the rate limiter's checks are already cached.

## Changes

Explicitly clear their cache in the base test setup

## How did you test this code?

opening this PR and seeing if 💣

### CODE REVIEW

File: posthog/test/base.py [hunk 1]
```diff
@@ -199,6 +200,9 @@ def setUp(self):
 
         # Clear the cached "is_cloud" setting so that it's recalculated for each test
         TEST_clear_cloud_cache()
+        # Clear the is_rate_limit lru_Caches so that they does not flap in test snapshots
```
Comment (pauldambra) [inline posthog/test/base.py:203]: this is the meat of it, the rest is noise caused by snapshots changing or needing to change expected query counts

File: posthog/test/base.py [hunk 2]
```diff
@@ -199,6 +200,9 @@ def setUp(self):
...
+        rate_limit.is_rate_limit_enabled.cache_clear()
+        rate_limit.get_team_allow_list.cache_clear()
```
Comment (pauldambra) [inline posthog/test/base.py:205]: this is the meat of it, the rest is noise caused by snapshots changing or needing to change expected query counts

**Comment by pauldambra** [comment]
fixes e.g. this https://github.com/PostHog/posthog/actions/runs/4974115694/jobs/8900365132?pr=15475

**Comment by neilkakkar** [comment]
Excellent work, thank you!

**Pull request merged**